docs/guides/plugin_dev/register_plugin.rst

```diff
@@ -1,4 +1,106 @@
 .. _register_plugins:
 
-Registering your plugins
-========================
+Implementing Custom behavior
+============================
+
+Due to the extremely wide range of data formats and conventions out in the world
+HydroMT offers a lot of ways to define custom behaviour, which is where you will
+probably spend most of your time as a plugin developer. We offer the possibility
+of extending what is in hydromt via the use of entry points. These entry points
+are a way of telling hydromt core about code you would like it to use to do specialised
+such as reading and writing a custom file format. Currently core exposes the following
+entry points to extend it's functionalities:
+
+* Model
+* ModelComponent
+* Resolvers
+* Driver
+* Catalog
+
+We will elaborate on each of these entrypoints in more detail in their own section but
+first we will cover the necessary information that is common to all: how to tell hydromt
+to use your custom behaviour. As an example we will use the hypothetical hydromt plugin
+for the (fictional) Advanced Water and Environmental Systems Optimization and Modeling
+Engine or AWESOME for short. The package will be called `hydromt_awesome`.
+
+EntryPoints
```

Comment:
Maybe also lean on the official (python main library) entrypoints documentation here